src/Nethermind/Nethermind.Blockchain/Synchronization/BlockDownloader.cs

```diff
@@ -157,6 +157,7 @@ public async Task<long> DownloadHeaders(PeerInfo bestPeer, int newBlocksToSkip,
 
                 if (headersSynced > 0)
                 {
+                    _blockTree.Flush();
```

Comment:
it introduces a lot of complexity to the state, can we flush straightaway or create a two level flush store where the questions are read form unflushed data?